PowerValidatedSolutions.psm1

```diff
@@ -16887,6 +17176,290 @@ Export-ModuleMember -Function Update-VcenterRootPasswordExpiration
+                    $WsaPasswordComplexityObject | Add-Member -notepropertyname "Min Numberic" -notepropertyvalue $WsaPasswordComplexity.minDigit
+                    $WsaPasswordComplexityObject | Add-Member -notepropertyname "Min Special" -notepropertyvalue $WsaPasswordComplexity.minSpecial
+                    $WsaPasswordComplexityObject | Add-Member -notepropertyname "Max Identical Adjacent" -notepropertyvalue $WsaPasswordComplexity.maxConsecutiveIdenticalCharacters
+                    $WsaPasswordComplexityObject | Add-Member -notepropertyname "History" -notepropertyvalue $WsaPasswordComplexity.History
+                } else {
+                    Write-Error "Unable to retrieve password complexity policy from Workspace ONE Access instance ($server): PRE_VALIDATION_FAILED"
+                }
+                return $WsaPasswordComplexityObject
+			}
+		}
+	} Catch {
+		Debug-ExceptionWriter -object $_
+	}
+}
+Export-ModuleMember -Function Request-WsaPasswordComplexity
+
+Function Request-WsaAccountLockout {
+	<#
+        .SYNOPSIS
+        Retrieves Workspace ONE Access account lockout
+
+        .DESCRIPTION
+        The Request-WsaAccountLockout cmdlet retrieves the Workspace ONE Access account lockout policy.
+        - Validates that network connectivity and authentication is possible to Workspace ONE Access
+        - Retrieve the account lockout policy
+
+        .EXAMPLE
+        Request-WsaAccountLockout -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1!
+        This example retrieves the account lockout policy for Workspace ONE Access instance sfo-wsa01
+    #>
+
+	Param (
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass
+	)
+	
+	Try {
+		if (Test-WsaConnection -server $server) {
+			if (Test-WsaAuthentication -server $server -user $user -pass $pass) {
+                if ($WsaAccountLockout = Get-WsaAccountLockout) {
+                    $WsaAccountLockoutObject = New-Object -TypeName psobject
+                    $WsaAccountLockoutObject | Add-Member -notepropertyname "FQDN" -notepropertyvalue $server
+                    $WsaAccountLockoutObject | Add-Member -notepropertyname "Max Failures" -notepropertyvalue $WsaAccountLockout.numAttempts
+                    $WsaAccountLockoutObject | Add-Member -notepropertyname "Unlock Interval (min)" -notepropertyvalue $WsaAccountLockout.unlockInterval
+                    $WsaAccountLockoutObject | Add-Member -notepropertyname "Failed Attempt Interval (min)" -notepropertyvalue $WsaAccountLockout.attemptInterval                   
+                } else {
+                    Write-Error "Unable to retrieve account lockout policy from Workspace ONE Access instance ($server): PRE_VALIDATION_FAILED"
```

Comment:
Suggestion to follow the established format:

```suggestion
                    Write-Error "Unable to retrieve Account Lockout Policy from Workspace ONE Access instance ($server): PRE_VALIDATION_FAILED"
```

Reply:
Fixed